### DISCUSSION TITLE
astroquery

### DISCUSSION THREAD

**Discussion by miraceti** [flet-dev/flet#3520] [Packages] [answered]
add astroquery to flet

**Comment by ndonkoHenri** [comment, marked as answer]
`astroquery` now works in Flet apps on iOS and Android.

It is itself is pure-Python, and so are `pyvo` and the rest of its dependency tree. The only pieces that needed cross-compiling were its two native dependencies — `pyerfa` (the ERFA fundamental-astronomy C library) and `astropy` — and both are now [supported](https://pypi.flet.dev/).

#### Notes
- `astropy` keeps a small config dir and a data cache, and it decides where at import time. Point them at Flet's app-storage directories before importing `astropy`/`astroquery` — config in the persistent data dir, cache in the temp dir:

```python
import os, tempfile

# astropy resolves these at import time, and only honors them if the dir exists.
_data = os.getenv("FLET_APP_STORAGE_DATA")                          # persistent
_temp = os.getenv("FLET_APP_STORAGE_TEMP") or tempfile.gettempdir() # cache/temp
_cfg   = os.path.join(_data or _temp, "astropy", "config")
_cache = os.path.join(_temp, "astropy", "cache")
os.makedirs(_cfg, exist_ok=True)
os.makedirs(_cache, exist_ok=True)
os.environ.setdefault("ASTROPY_CONFIG_DIR", _cfg)
os.environ.setdefault("ASTROPY_CACHE_DIR", _cache)

import astroquery  # ...your imports
``` 

[`FLET_APP_STORAGE_DATA`](https://flet.dev/docs/reference/environment-variables/#flet_app_storage_data) / [`FLET_APP_STORAGE_TEMP`](https://flet.dev/docs/reference/environment-variables/#flet_app_storage_temp) are set by Flet before your Python runs. On iOS this is optional since `HOME` is already a writable sandbox, but it's the portable choice across Android too.

- **Offline data**: IERS Earth-orientation data is bundled (`astropy-iers-data`), so `Time`/coordinate work runs offline. The `astroquery` lookups themselves query remote services, so those need network access.
- **Credentials**: there's no system keyring backend on mobile, so anonymous queries (SIMBAD, VizieR, Gaia TAP, …) work out of the box; services that rely on the OS keyring for stored logins won't.
